presto-main/src/main/java/com/facebook/presto/server/protocol/QueuedStatementResource.java

```diff
@@ -484,7 +488,10 @@ public synchronized QueryResults getInitialQueryResults(UriInfo uriInfo, String
             // Hence it is safe to hardcode the token to be 0.
             return transform(
                     query.waitForResults(0, uriInfo, getScheme(xForwardedProto, uriInfo), maxWait, TARGET_RESULT_SIZE),
-                    results -> QueryResourceUtil.toResponse(query, results, compressionEnabled),
+                    results -> {
+                        query.getTracer().addPoint("Result is ready, redirecting to execution endpoint to fetch result data");
+                        return QueryResourceUtil.toResponse(query, results, compressionEnabled);
+                    },
```

Comment:
Same as the question above. Do we need to track all HTTP requests (which could be thousands of them) back to client?